public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx

```diff
@@ -1,226 +1,769 @@
+    useGetLabelsFromDataSourceName(GRAFANA_RULES_SOURCE_NAME);
+
+  // Create label options for the multi-select dropdown
+  const labelOptions = useMemo((): Array<ComboboxOption<string>> => {
+    const infoOption: ComboboxOption<string> = {
+      label: t('label-dropdown-info', "Can't find your label? Enter it manually"),
+      value: '__GRAFANA_LABEL_DROPDOWN_INFO__',
+      infoOption: true,
+    };
+
+    const selectableOptions = Array.from(grafanaLabels.entries())
+      .flatMap(([key, values]) =>
+        Array.from(values).map((value: string) => {
+          const labelPair = `${key}=${value}`;
+          return { label: labelPair, value: labelPair };
+        })
+      )
+      .sort((a, b) => new Intl.Collator().compare(a.label, b.label));
+
+    return [...selectableOptions, infoOption];
+  }, [grafanaLabels]);
+
+  // Generate appropriate placeholder text
+  const namespacePlaceholder = useMemo(() => {
+    if (isLoadingNamespaces) {
+      return t('common.loading', 'Loading...');
+    }
+    if (namespaceOptions.length === 0) {
+      return t('alerting.rules-filter.no-namespaces', 'No folders available');
+    }
+    return t('alerting.rules-filter.filter-options.placeholder-namespace', 'Select namespace');
+  }, [isLoadingNamespaces, namespaceOptions.length]);
+
+  const groupPlaceholder = useMemo(() => {
+    if (isLoadingNamespaces) {
+      return t('common.loading', 'Loading...');
+    }
+    if (allGroupNames.length === 0) {
+      return t('alerting.rules-filter.no-groups', 'No groups available');
+    }
+    return t('grafana.select-group', 'Select group');
+  }, [isLoadingNamespaces, allGroupNames.length]);
+
+  // turn the filterState into form default values
+  const { handleSubmit, reset, register, control } = useForm<AdvancedFilters>({
+    defaultValues,
+  });
+
+  // Update form values when filterState changes (e.g., when popup reopens)
+  useEffect(() => {
+    // Skip if we're in the middle of a manual reset
+    if (isManualResetRef.current) {
+      isManualResetRef.current = false;
+      return;
+    }
+
+    const newDefaultValues = searchQueryToDefaultValues(filterState);
+    reset(newDefaultValues);
+  }, [filterState, reset]);
+
+  const submitAdvancedFilters = handleSubmit(onSubmit);
+
+  const dataSourceOptions = useMemo(() => {
+    return getDataSourceSrv()
+      .getList({ alerting: true })
+      .map((ds: DataSourceInstanceSettings) => ({ label: ds.name, value: ds.name }));
+  }, []);
 
   return (
-    <Stack direction="column" gap={0}>
-      <Label>
-        <Trans i18nKey="common.search">Search</Trans>
-      </Label>
-      <Stack direction="row">
-        <Input prefix={filterOptions} />
+    <form
+      onSubmit={submitAdvancedFilters}
+      onReset={() => {
+        isManualResetRef.current = true;
+        reset(emptyAdvancedFilters);
+        trackFilterButtonClearClick();
+        onClear();
+      }}
+    >
+      <Stack direction="column" alignItems="end" gap={2}>
```

Comment:
How about splitting this into multiple components and using `useFormContext` to easily access the form?